twill-yarn/src/test/java/org/apache/twill/filesystem/HDFSLocationTest.java

```diff
@@ -47,4 +52,14 @@ public static void finish() {
   protected LocationFactory createLocationFactory(String pathBase) throws Exception {
     return new HDFSLocationFactory(dfsCluster.getFileSystem(), pathBase);
   }
+
+  // TODO (TWILL-209): figure out how to make MiniDFSCluster enforce permissions
+  @Ignore
+  @Test(expected = Exception.class)
+  public void testPermissions() throws IOException {
+    // create a directory that does not permit anything
+    dfsCluster.getFileSystem().mkdir(new Path("/a"), FsPermission.valueOf("----------"));
```

Comment:
Good point. Anyway, this test does not throw an exception and is ignored because of that. I guess this is a typical problem with using expected=, we can never know whether it is thrown by the line in the code where we expect. So should we rewrite all our test cases not to use that? And explicitly fail if the exception is not thrown?